crates/distribution-types/src/file.rs

```diff
@@ -51,7 +53,12 @@ impl File {
             size: file.size,
             upload_time_utc_ms: file.upload_time.map(|dt| dt.timestamp_millis()),
             url: if file.url.contains("://") {
-                FileLocation::AbsoluteUrl(file.url)
+                let url = safe_copy_url_auth(
+                    base,
+                    Url::parse(&file.url)
+                        .map_err(|err| FileConversionError::Url(file.url.clone(), err))?,
+                );
+                FileLocation::AbsoluteUrl(url.to_string())
```

Comment:
I think we intentionally changed this to use a string rather than a URL because we saw a huge regression due to excessive URL parsing, serialization, and deserialization. We should at least benchmark this since we're now doing an extra URL parse and serialization each time.

Reply:
😬

Reply:
Benches look fine